### DISCUSSION TITLE
Issues with language server config on latest helix

### DISCUSSION THREAD

**Discussion by dryya** [helix-editor/helix#7713] [Q&A] [answered]
Hi there. I recently started using `helix-dev` and updated my language config according to the new format that landed with multi-LS support.

Here is my languages.toml: https://gist.github.com/dryya/c259790e7ef18ed0c9ee3e67b288e1cd

I can confirm that the config is being passed to the language servers, because when I added `texlab.args = [ "-vvvv" "--log-file" "/tmp/texlab.log" ]` I can see texlab reporting

```
DEBUG - < {"jsonrpc":"2.0","method":"initialize", [...] ,"clientInfo":{"name":"helix","version":"23.05"},"initializationOptions":{"build":{"forwardSearchAfter":true,"onSave":true},"chktex":{"onEdit":true},"forwardSearch":{"args":["--synctex-forward","%l:1:%f","%p"],"executable":"zathura"}}, [...]}
```

However, on 23.05 with the old config, `.tex` files would compile on save like they're supposed to with this config, but on the latest master they don't. I feel like it's more likely that I'm doing something wrong than that I have a bug, but I'm stumped.

I also tested with nil, and with this config, `nixfmt` isn't run on save, while with my old config, it was. For reference, the old config was

```
[[language]]
auto-format = true
name = "nix"

[config.nil.formatting]
command = ["/nix/store/ihdv735ihacv6f2l83h22rcii3v2yd5b-nixfmt-0.5.0-bin/bin/nixfmt", "-q"]
```

**Comment by kirawi** [comment, marked as answer]
It should be `[language-server.nil.config]`. These also have to be at the top of the file.

**Reply by dryya** [reply]
My config is home-manager generated, so I'll raise an issue there about putting the [language-server] table at the top of the file. However, my issue was actually unrelated: I noticed that both nil and texlab store their configuration options in a top-level table with the name of the server, so I needed to change to:

```
[language-server.texlab.config.texlab.build]
forwardSearchAfter = true
[language-server.nil.config.nil]
formatting = { command = ["nixfmt"] }
```

I realize now that `[config.nil.formatting]` from my 23.05 config has that table, but I got confused when first reading it and thought it just referred to the name of the language server.

**Reply by dryya** [reply]
Actually, having them at the top doesn't seem to be necessary as far as I can tell, since everything is working perfectly with the following (even if it looks a little odd):
```
[[language]]
auto-format = true
name = "nix"

[language-server]
[language-server.nil]
command = "/nix/store/pl73ayc481hi2plkfw0cpm4csvjmjhbf-nil-2023-05-09/bin/nil"

[language-server.nil.config.nil.formatting]
command = ["/nix/store/ihdv735ihacv6f2l83h22rcii3v2yd5b-nixfmt-0.5.0-bin/bin/nixfmt", "-q"]
```

**Reply by gabydd** [reply]
Yeah it's not necessary there are some weird things with reloading config and multiple language-servers but other then that it should be fine
